### ISSUE TITLE
Replication controller to re-schedule Pod in `ContainerCreating` and `ImageNotReady` status for a long time

### ISSUE THREAD

**Issue by xinxiaogang** [kubernetes/kubernetes#21657]
I have meet an issue in my cluster described here https://github.com/kubernetes/kubernetes/issues/21656.

My question is, my Pod in a RC was scheduled to a node which docker CIDR already full. So the Pod stay in `ImageNotReady` status. Only after I delete the Pod manually by `kubectl delete pod`, the Pod got rescheduled to another node.

I think from RC perspective, some loop should take the responsibility on Pod status monitoring. If the Pod can not be `Running` for sometime, if the failure reason is not related with Pod itself, the RC should kill the Pod and reschedule to another node.

For example, if the Pod stays in `ContainerCreating` status for a long time which means something wrong with the node's docker, the Pod should be rescheduled.

**Comment by xinxiaogang** [comment]
Anyone to take a look at this?

**Comment by quinton-hoole** [comment]
Agreed, assigning priority p1.  We should try to fit this into our next release.

**Comment by balexx** [comment]
+1
Just came across this when human error messed up docker data dir permissions.

**Comment by xinxiaogang** [comment]
What release will this feature be implemented ?

**Comment by kargakis** [comment]
@kubernetes/sig-scheduling-feature-requests is this something that should be handled by the rescheduler?

**Comment by resouer** [comment]
@kargakis  One thing I am not sure is, `ContainerCreating` can also mean sth wrong on node, then keeping the Pod there will help user to debug the failure. Is this make sense to you?

**Comment by dims** [comment]
@quinton-hoole @xinxiaogang Looks like there's nothing here that can be done for 1.8. Can we please move this out of 1.8?
